src/main/java/org/jetbrains/bsp/bazel/server/bsp/BazelBspServerBuildManager.java

```diff
@@ -144,7 +145,7 @@ public BuildTarget getBuildTargetForRule(Build.Rule rule) {
         .collect(Collectors.toList());
   }
 
-  private Optional<ScalaBuildTarget> getScalaBuildTarget() {
+  private Optional<ScalaBuildTarget> getScalaBuildTarget(Build.Rule rule) {
```

Comment:
This rule is passed around so much since the rule holds most of the information regarding the current target. I think we need a specific class that holds all these getters.